File: ios/DeviceCapabilitiesService.m
//
//  DeviceCapabilitiesService.m
//  heartsapp
//
//  Created by havarma on 28/03/20.
//  Copyright © 2020 Facebook. All rights reserved.
//

#import <Foundation/Foundation.h>
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(DeviceCapabilitiesService, NSObject)


RCT_EXTERN_METHOD(hasDoNotDisturbPermission:(nonnull NSDictionary *)input
                  resolver:(RCTPromiseResolveBlock)resolve
                  rejecter:(RCTPromiseRejectBlock)reject
                  )

RCT_EXTERN_METHOD(requestDoNotDisturbPermission:(nonnull NSDictionary *)input
                  resolver:(RCTPromiseResolveBlock)resolve
                  rejecter:(RCTPromiseRejectBlock)reject
                  )

RCT_EXTERN_METHOD(enableDoNotDisturbMode:(nonnull NSDictionary *)input
                  resolver:(RCTPromiseResolveBlock)resolve
                  rejecter:(RCTPromiseRejectBlock)reject
                  )


RCT_EXTERN_METHOD(disableDoNotDisturbMode:(nonnull NSDictionary *)input
                  resolver:(RCTPromiseResolveBlock)resolve
                  rejecter:(RCTPromiseRejectBlock)reject
                  )

@end

File: ios/DeviceCapabilitiesService.swift
//
//  DeviceCapabilities.swift
//  heartsapp
//
//  Created by havarma on 28/03/20.
//  Copyright © 2020 Facebook. All rights reserved.
//

import Foundation

@objc(DeviceCapabilitiesService)
class DeviceCapabilities: NSObject {
  @objc static func requiresMainQueueSetup() -> Bool {
      return false
    }
  
  @objc func hasDoNotDisturbPermission(
  _ input: NSDictionary,
  resolver resolve: @escaping RCTPromiseResolveBlock,
  rejecter reject: @escaping RCTPromiseRejectBlock)  {
    resolve(true)
  }
  
  @objc func requestDoNotDisturbPermission(
  _ input: NSDictionary,
  resolver resolve: @escaping RCTPromiseResolveBlock,
  rejecter reject: @escaping RCTPromiseRejectBlock)  {
    resolve(true)
  }
  
  @objc func enableDoNotDisturbMode(
  _ input: NSDictionary,
  resolver resolve: @escaping RCTPromiseResolveBlock,
  rejecter reject: @escaping RCTPromiseRejectBlock)  {
    resolve(true)
  }
  
  @objc func disableDoNotDisturbMode(
  _ input: NSDictionary,
  resolver resolve: @escaping RCTPromiseResolveBlock,
  rejecter reject: @escaping RCTPromiseRejectBlock)  {
    resolve(true)
  }
}
